clients/da-vinci-client/src/test/java/com/linkedin/davinci/replication/merge/TestMergeWithValueLevelTimestamp.java

```diff
@@ -74,7 +76,8 @@ public void testPut() {
     // verify same timestamp case
     mergeConflictResult = mergeConflictResolver.put(
         Lazy.of(() -> oldBB),
-        Optional.of(new RmdWithValueSchemaId(1, RMD_VERSION_ID, GenericData.get().deepCopy(rmdSchemaV1, rmdRecord))),
```

Comment:
Could we get rid of Optionals from the merge conflict resolver API? I don't think it provides any value. Nullable params would work just as well.

Reply:
I think Optional gives us a sense to force checking the existence of the value, so it is less error prone compared to raw value and null check?

Reply:
Bah... I think it's a false sense of safety, since Optionals can still be null too, so the really safe thing would be to check that it's not null and present... but that seems pretty clumsy. See the post linked from the style guide to read on that.

More importantly though, conflict resolution is a hot path operation, so I'd prefer making it efficient. Even coming from someone who considers optionals to be actually good for maintainability, they arrive at the conclusion that they're very inefficient: https://pkolaczk.github.io/overhead-of-optional/

Reply:
How about we do it in a separate RB? I feel like there is a lot of related changes to be made as this optional concept has been deeply rooted around the MergeConflictResolver implementation. This PR has already touched 20 files and it would be great not to make more here... I can cut a JIRA ticket to track this change.

Reply:
Sure. I thought of it sonce some of the relevant lines are already touched by this PR, but doing it would increase the scope further, of course. As you wish.